### PR TITLE
1018 create the android call quality guide for ma

### PR DESCRIPTION
**Pull request by saudsami**

### CODE REVIEW

File: media-accelerator/get-started/get-started-sdk.mdx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+type: docs
+description: >
+  Rapidly develop and easily enhance IoT solutions with audio communication and face-to-face interaction.
+---
+
+import GetStartedSDK from '@docs/shared/media-accelerator/get-started/_get-started-sdk.mdx';
+
+export const toc = [{}];
+
+<GetStartedSDK  product="Media Accelerator" company="Agora" sdk="RTSA SDK" client="app" />
```
Comment (billy-the-fish) [inline media-accelerator/get-started/get-started-sdk.mdx:13]: Media Accelerator SDK. 

Don't we have global variables for this?

File: shared/media-accelerator/develop/_ensure-channel-quality.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  - Opus
+  - G722
+  - G711A (PCMU) 
+  - G711U (PCMU) 
+
+  You specify an audio codec when you join a channel. You also set the sampling rate and the number of channels at this time, depending on your audio quality requirements. If your <Vpl k="CLIENT"/> needs a custom encoding algorithm, you disable use of a built-in codec and implement your own encoder. 
+
+- **Adjust sending bit rate in real time**
+
+  In order to optimize data transmission and avoid network congestion, <Vpd k="SDK"/> suggests adjusting the sending bit rate in real-time according to changes in network conditions.
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/_ensure-channel-quality.mdx:30]: The SDK suggests? 

Best practice is to ...

File: shared/media-accelerator/develop/_ensure-channel-quality.mdx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import ProjectTest from '@docs/shared/media-accelerator/develop/ensure-channel-quality/project-test/index.mdx';
+import Reference from '@docs/shared/media-accelerator/develop/ensure-channel-quality/reference/index.mdx';
+
+<Introduction />
+
+This page shows you how to use <Vpd k="SDK"/> features to account for these factors and ensure optimal audio and video quality in your {props.PRODUCT} <Vpl k="CLIENT" />.
+
+## Understand the tech
+
+<Vpd k="SDK"/> provides the following features to deal with channel quality issues:
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/_ensure-channel-quality.mdx:15]: to provide the best audio and video quality in your < variable for app>.

File: shared/media-accelerator/develop/_ensure-channel-quality.mdx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+- **Choose an audio codec to optimize bit rate and quality**
+
+  In audio and video streaming, the choice of an encoding algorithm affects both quality and bit rate. Your goal is to lower the required bit rate while maintaining quality at the desired level. <Vpd k="SDK" /> offers the following built-in audio codecs to choose from:
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/_ensure-channel-quality.mdx:19]: following built-in codec:

File: shared/media-accelerator/develop/_ensure-channel-quality.mdx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  - Opus
+  - G722
+  - G711A (PCMU) 
+  - G711U (PCMU) 
+
+  You specify an audio codec when you join a channel. You also set the sampling rate and the number of channels at this time, depending on your audio quality requirements. If your <Vpl k="CLIENT"/> needs a custom encoding algorithm, you disable use of a built-in codec and implement your own encoder.
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/_ensure-channel-quality.mdx:26]: You specify an audio codec when you join a channel. Depending on your audio quality requirements  you also set the sampling rate and the number of channels.

\<why you do it> \<how you do it>

File: shared/media-accelerator/develop/_ensure-channel-quality.mdx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+  When a user changes the transmission status of the local audio or video stream, the <Vpd k="SDK"/> triggers a corresponding callback to prompt the remote user to suspend or resume sending their audio or video stream.
+
+- **Log files** 
+
+  <Vpd k="SDK"/> provides configuration options that you use to customize the location, content and size of log files containing key data of <Vpd k="SDK"/> operation. When you set up logging, <Vpd k="SDK"/> writes information messages, warnings, and errors regarding activities such as initialization, configuration, connection and disconnection to log files. Log files are useful in detecting and resolving channel quality issues.
+
+The following figure shows the workflow you need to implement to ensure channel quality in your <Vpl k="CLIENT"/>:
+
+![Ensure Channel Quality](/images/media-accelerator/media-accelerator-channel-quality.svg)
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/_ensure-channel-quality.mdx:46]: Why are you saying service here when we are talking about an app?

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+        android:id="@+id/MuteLocalVideo"
+        android:text="Mute local video"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/MuteLocalAudio"
+        android:layout_alignStart="@id/JoinButton" />
+    ```
+
+### Handle the system logic
+
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx:26]: Intro sentence.

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    In `MainActivity.java`, add the following to the list of `import` statements:
+
+    ```java
+    import java.lang.Math;
+    import android.widget.CheckBox;
+    import android.widget.CompoundButton;
+    ```
+
+1. **Define variables to access checkboxes**
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx:37]: the checkboxes

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+    In `/app/java/com.example.<projectname>/MainActivity`, add the following declarations to the `MainActivity` class:
+
+    ```java
+    private CheckBox checkMuteLocalAudio;
+    private CheckBox checkMuteLocalVideo;
+    ```
+
+### Implement channel quality features
+
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx:47]: Intro sentence

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+1. **Configure the <Vpd k="SDK" /> log file**
+
+    To customize the location, and content of the log file, add the following code to `setupAgoraRtcService()` before `agoraRtcService.init(appId, agoraRtcEvents, options);`
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx:50]: Ok, you will need to go back to the quickstart for this, but AgoraRtcService is the class, but we call it agoraEngine for all the other platforms, let's be consistent.

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+    ```java
+    channelOptions.audioCodecOpt.audioCodecType
+            = AgoraRtcService.AudioCodecType.AUDIO_CODEC_TYPE_OPUS;
+    channelOptions.audioCodecOpt.pcmSampleRate = 16000;
+    channelOptions.audioCodecOpt.pcmChannelNum = 1;
+    ```
+  
+    Choose from `OPUS`, `G722`, `G711A`, and `G711U` built-in audio codecs in the <Vpd k="SDK" />. 
+    
+    To use your own audio codec, set `channelOptions.audioCodecOpt.audioCodecType` to `AUDIO_CODEC_DISABLED`. When you call `agoraRtcService.sendAudioData`, set the `dataType` parameter of `AudioFrameInfo` to your encoding format. This setting just transmits the audio encoding format to the receiving end. When you disable use of a built-in audio codec, <Vpd k="SDK" /> does not process the audio. The receiving end obtains the encoded audio data and the encoding format in the `onAudioData` callback and decodes it using a custom decoder.
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx:71]: This setting transmits the ...


Avoid words like just or simply

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+
+            lastResolutionLevel = curResolutionLevel;
+            lastTargetBitrate = curTargetBitrate;
+        }
+    }
+    ```
+
+1. **Manage audio and video streaming status**
+
+    When a user taps a Checkbox, you pause or resume sending local audio and video streams. To do this, add the following code to the `onCreate` method in the `MainActivity` class:
```
Comment (billy-the-fish) [inline shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx:131]: either checkbox a term or `Checkbox` an object.

File: shared/video-sdk/authentication-workflow/reference/android.mdx [hunk 13]
```diff
@@ -1,9 +1,17 @@
 <PlatformWrapper platform="android">
-        
+
 ### API reference
 
+<ProductWrapper notAllowed="media-accelerator">
```
Comment (billy-the-fish) [inline shared/video-sdk/authentication-workflow/reference/android.mdx:5]: We need to track down the API ref files.

**Review by billy-the-fish — CHANGES_REQUESTED** [review]
Good work, nearly there.

File: shared/media-accelerator/develop/_ensure-channel-quality.mdx (hunk 2 above)
Comment (saudsami) [reply shared/media-accelerator/develop/_ensure-channel-quality.mdx]: fixed

File: shared/media-accelerator/develop/_ensure-channel-quality.mdx (hunk 3 above)
Comment (saudsami) [reply shared/media-accelerator/develop/_ensure-channel-quality.mdx]: fixed

File: shared/media-accelerator/develop/_ensure-channel-quality.mdx (hunk 5 above)
Comment (saudsami) [reply shared/media-accelerator/develop/_ensure-channel-quality.mdx]: fixed

File: shared/media-accelerator/develop/_ensure-channel-quality.mdx (hunk 6 above)
Comment (saudsami) [reply shared/media-accelerator/develop/_ensure-channel-quality.mdx]: Changed service to engine

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx (hunk 7 above)
Comment (saudsami) [reply shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx]: Added

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx (hunk 8 above)
Comment (saudsami) [reply shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx]: fixed

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx (hunk 9 above)
Comment (saudsami) [reply shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx]: Added

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx (hunk 10 above)
Comment (saudsami) [reply shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx]: Done

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx (hunk 11 above)
Comment (saudsami) [reply shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx]: removed

File: shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx (hunk 12 above)
Comment (saudsami) [reply shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx]: Fixed

File: shared/video-sdk/authentication-workflow/reference/android.mdx (hunk 13 above)
Comment (saudsami) [reply shared/video-sdk/authentication-workflow/reference/android.mdx]: I found API references on the Chinese website:
https://docs.agora.io/cn/RTSA/API%20Reference/rtsa_c/index.html
https://docs.agora.io/cn/RTSA/API%20Reference/rtsa_java/classio_1_1agora_1_1rtc_1_1_agora_rtc_service.html

File: media-accelerator/get-started/get-started-sdk.mdx (hunk 1 above)
Comment (saudsami) [reply media-accelerator/get-started/get-started-sdk.mdx]: Removed.

**Pull request merged**